src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5216,6 +5216,8 @@ TorrentImpl *SessionImpl::createTorrent(const lt::torrent_handle &nativeHandle,
 
     if (isRestored())
     {
+        torrent->saveResumeData(lt::torrent_handle::save_info_dict);
```

Comment:
General question:
So now we save the info_dict regardless if the torrent is a magnet or not?
Do we still save the .torrent files?

Reply:
> So now we save the info_dict regardless if the torrent is a magnet or not?
Do we still save the .torrent files?

Metadata (aka info section) is still saved separately from other data in its own .torrent file however this file cannot be treated as a real torrent file generally.

Reply:
`saveResumeData()` is traditionally named so by analogy with the wrapped method of `libtorrent::torrent_handle` (although I feel some naming used in libtorrent confusing). It just generates resume data. So here we tell it to include metadata there as well (of course, if they are available for the current torrent, i.e. if it is added from a .torrent file, or if the metadata was preloaded for the magnet link).
Resume data storage subsystem will extract metadata and store it separately.

Reply:
>Resume data storage subsystem will extract metadata and store it separately.

Oh this is what I was missing.
I knew that `saveResumeData()` would save resumedata which would include the metadata. I didn't know if we still(and how) saved the metadata separately.